pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -55,21 +58,31 @@ func AddToManager(mgr controllerruntime.Manager, opts Options) error {
 	if err := opts.CIOperatorConfigAgent.AddIndex(configIndexName, configIndexFn); err != nil {
 		return fmt.Errorf("failed to add indexer to config-agent: %w", err)
 	}
-
+	configChangeChannel, err := opts.CIOperatorConfigAgent.SubscribeToIndexChanges(configIndexName)
```

Comment:
It is a very good question. Here is my understanding:
The events in this channel is not K8S events. They are from watching the changes on the ci-op's configuration files on the file system.
If the system is restarted, it will treat all promoted tags as new ones.
Then we check all tags on the cluster, if any of them does not exist, we reconcile.